src/lib/io/include/sol-iio.h

```diff
@@ -77,6 +77,7 @@ typedef struct sol_iio_config {
     const void *data; /**< User defined data to be sent to sol_iio_reader_cb */
     int buffer_size; /**< The size of reading buffer. 0: use device default; -1: disable buffer and readings will be performed on channel files on sysfs. */
     int sampling_frequency; /**< Device sampling frequency. -1 uses device default */
+    char sampling_frequency_name[PATH_MAX];
```

Comment:
Suggest to add comment like other attribute.
Some drivers exposing its sampling frequency that is shared by channel type. For e.g: in_magn_sampling_frequency, in_accel_sampling_frequency.